### PR TITLE
[RFR] Refactor (once again) the make script

### PR DESCRIPTION
**Pull request by valeriangalliat**
This time use [Jake](http://jakejs.com/).

* We gain the ability to render a specific theme for the gallery, or refresh a specific screenshot without an environment "hack" like with previous `make.js`.

* Instead of a single bloated file, tasks are grouped logically in different files in `jakelib` directory.

* Make the output fabulous. <sup><sub>Background and prompt not included.</sub></sup>

  Example of output

* Use Webshot again, in preparation for PhantomJS 2.0 Linux release. This is because PhantomJS don't get sometimes stuck in an `epoll_wait` call unlike atom-shell and node-webkit, but PhantomJS 1.0 can't rander SVGs properly, and this bug is fixed in the 2.0 release.

**Note:** this PR should be updated once PhantomJS 2.0 is available as a Node module, and then we'll be able to merge it.

* [x] <del>Wait [PhantomJS 2.0](http://phantomjs.org/download.html) npm release.</del> It's been more than one year, let's use something else maybe.
* [ ] Update screenshots and previews.
* [ ] Merge.

### CODE REVIEW

**Comment by HugoGiraudel** [comment]
Is this still relevant?

**Comment by valeriangalliat** [comment]
Damned yeah. Still waiting for PhantomJS 2.0 static build. :trollface:

**Comment by pascalduez** [comment]
I added 2 new themes, overall worked okayish. Except the default thumb has a weird sidebar.
Also there's a new theme that does not produce an index.html, so had to generate one by hand with pandoc.

Other themes I haven't looked at yet:
https://github.com/DigitalInnovation/sassdoc-theme-mns
https://github.com/LPLabs/sassdoc-theme-lonelyplanet
https://github.com/allanhortle/bruce-theme
https://github.com/CasperSleep/sassdoc-theme-nightshade

**Comment by valeriangalliat** [comment]
@SassDoc/owners Moved to electron-screenshot-service. Looks good.

Only annoying thing is https://github.com/FWeinb/electron-screenshot-service/issues/7 but we can fix this later.

Also the Vulcan theme is outdated and cannot compile with SassDoc 2, and the Rest theme doesn't have an `index.html` so the build script doesn't know what to screenshot.

There wasn't any issue when you added the Rest theme @pascalduez?

**Comment by pascalduez** [comment]
@valeriangalliat For the Rest theme, I had to do it by hand as you can imagine :) So generating the index.html with Pandoc. Maybe you could leave it untouched ? 
Also for  the Vulcan one, I removed it from the list of theme "themes" task, so to keep the old files.
But indeed, should we keep it in the list then ?

**Comment by pascalduez** [comment]
There's a new kid on the block: https://github.com/oddbird/sassdoc-theme-herman

File: jakelib/gallery.jake [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  .map(x => { x.name = x.image.replace(/\..*$/, ''); return x })
+
+const galleryDir = 'assets/images/gallery'
+
+directory(galleryDir)
+
+gallery.forEach(item => {
+  const image = `${galleryDir}/${item.image}`
+
+  file(image, [galleryDir, 'screenshot'], async () => {
```
Comment (HugoGiraudel) [inline jakelib/gallery.jake]: BAM BAM ASYNC/AWAIT IS IN DA PLACE YO BITCHES!

**Comment by HugoGiraudel** [comment]
This is awesome. :100:

File: jakelib/gallery.jake (hunk 1 above)
Comment (pascalduez) [inline jakelib/gallery.jake]: Did you know it's been [in core](https://github.com/SassDoc/sassdoc/blob/master/src/sassdoc.js#L129) since 2.0 ? ;)

Comment (HugoGiraudel) [inline jakelib/gallery.jake]: Fuck, I don’t know this project anymore… :(

File: package.json [hunk 2]
```diff
@@ -1,26 +1,19 @@
 {
-  "dependencies": {
-    "array-zip": "^1.0.0",
-    "atom-screenshot": "^0.4.3",
-    "babel": "^4.4.0",
+  "devDependencies": {
+    "babel": "^4.4.3",
```
Comment (pascalduez) [inline package.json]: 5, or 6 ?

File: package.json [hunk 3]
```diff
@@ -1,26 +1,19 @@
 {
-  "dependencies": {
-    "array-zip": "^1.0.0",
-    "atom-screenshot": "^0.4.3",
```
Comment (pascalduez) [inline package.json]: `electron-screenshot-service` not in there ?

**Comment by pascalduez** [comment]
Just realized this is a PR so :+1:  !

File: jakelib/gallery.jake (hunk 1 above)
Comment (valeriangalliat) [inline jakelib/gallery.jake]: :laughing:

File: package.json (hunk 2 above)
Comment (valeriangalliat) [inline package.json]: Need and upgrade yes! But now it works, still... gonna have a look later.

File: package.json (hunk 3 above)
Comment (valeriangalliat) [inline package.json]: Like Webshot, it's downloaded only the first time a task needs to take a shot. Avoids long `npm i` if you just need to update the changelog or whatever.

File: package.json (hunk 2 above)
Comment (pascalduez) [inline package.json]: If it works, don't touch anything :)

File: package.json (hunk 3 above)
Comment (pascalduez) [inline package.json]: Oh right, now I remember, happy times :)

**Comment by valeriangalliat** [comment]
Updated all dependencies, switched to ES6 imports. Merging.

**Pull request merged**